lib/getopt/getopt.c

```diff
@@ -0,0 +1,973 @@
+/* Getopt for Microsoft C
+This code is a modification of the Free Software Foundation, Inc.
+Getopt library for parsing command line argument the purpose was
+to provide a Microsoft Visual C friendly derivative. This code
+provides functionality for both Unicode and Multibyte builds.
+
+Date: 02/03/2011 - Ludvik Jerabek - Initial Release
+Version: 1.0
+Comment: Supports getopt, getopt_long, and getopt_long_only
+and POSIXLY_CORRECT environment flag
+License: LGPL
```

Comment:
> OTOH vcpkg includes a getopt-win32 package. Why not use that?

That's exactly what [6228bd8](https://github.com/libofx/libofx/pull/63/commits/6228bd8ce3ee9e65cf63be1b18a843eea7a6eadc) does, but it's a WIP.

Reply:
I mean make the getopt-win32 package a dependency when building with MSVC instead of building it in here.

Reply:
> I mean make the getopt-win32 package a dependency when building with MSVC

Not sure how you think this would work, but `vcpkg` is not part of `MSVC`. Relying on `vcpkg` would mean compilation with pure MSVC wouldn't be possible. Refer to my first comment in this PR for context.

Reply:
Your original comment raises the same points but doesn't elaborate on the why's .  I'm trying to elicit the reasoning. Let's start with is compiling a stand-alone libofx with MSVC a worthwhile goal and if so, why? Do you know of some GPLV2 compliant project that compiles with only MSVC and might benefit from libofx?

If that's the case there's a fourth alternative to your three: Write a cmake FindGetoptWin32.cmake and let the user who wants to build on MSVC get the code and build it separately.

Reply:
> You may need to contact the author to find out what version that is

The [vcpkg item](https://vcpkg.info/port/getopt-win32) points to a [GitHub repository](https://github.com/libimobiledevice-win32/getopt) that points to [the original article](https://www.codeproject.com/Articles/157001/Full-getopt-Port-for-Unicode-and-Multibyte-Microso) that has at the top a link to [the license](http://www.opensource.org/licenses/lgpl-3.0.html). It's LGPLV3. That might be a deal breaker for @benoitg because including that source here [makes libofx GPLV3](https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility) instead of the current GPLV2+.

Reply:
> > You may need to contact the author to find out what version that is
> 
> The [vcpkg item](https://vcpkg.info/port/getopt-win32) points to a [GitHub repository](https://github.com/libimobiledevice-win32/getopt) that points to [the original article](https://www.codeproject.com/Articles/157001/Full-getopt-Port-for-Unicode-and-Multibyte-Microso) that has at the top a link to [the license](http://www.opensource.org/licenses/lgpl-3.0.html). It's LGPLV3. That might be a deal breaker for @benoitg because including that source here [makes libofx GPLV3](https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility) instead of the current GPLV2+.

Indeed. The alternative is MIT-licensed https://github.com/takamin/win-c

Reply:
> Let's start with is compiling a stand-alone libofx with MSVC a worthwhile goal and if so, why?

Because of the tools provided by the libofx itself (ofxdump, ofxconnect, ofx2qif). The former two are, after all, the __only__ components that depend on `getopt`. If the library itself was to be considered, we wouldn't be discussing this at all.

> if that's the case there's a fourth alternative to your three: Write a cmake FindGetoptWin32.cmake and let the user who wants to build on MSVC get the code and build it separately.

Indeed, that's basically the "we can just do nothing" option. Whether a user decides to build their own copy or resorts to vcpkg, it would be up to them. In such case, I would convert the `getopt/CMakeLists.txt` to `FindGetoptWin32.cmake`.

Reply:
> It's LGPLV3. That might be a deal breaker for @benoitg because including that source here [makes libofx GPLV3](https://www.gnu.org/licenses/gpl-faq.html#AllCompatibility) instead of the current GPLV2+.

Actually, looking at the source code, the license is "either version 2 of the License, or (at your option) any later version. ", which you already noticed by referring to it as GPLV2**+**. That makes libofx is GPL3 compatible, which, in turn, is compatible with LGPLV3, so I don't think we have a problem here?